doc: Adds example for GCP in `mongodbatlas_cloud_provider_access` resources and `mongodbatlas_encryption_at_rest` resource (#3645)
mongodb/terraform-provider-mongodbatlas: examples/mongodbatlas_cloud_provider_access/gcp/README.md

> # MongoDB Atlas Cloud Provider Access with Google Cloud Platform (GCP)

Reviewer: Can you also take an action item to add links to these new examples in the 2.0 branch once this is released? under "Further Examples"

Author: not sure I understand. this will eventually be in the 2.0 branch when master is merged to the 2.0 or when 2.0 branch is merged to master

Author: Do you mean links to the examples in the resources?

Reviewer: so in this PR https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3632/files# we added links to all examples in the resource docs

This example demonstrates how to set up MongoDB Atlas Cloud Provider Access with Google Cloud Platform (GCP). This creates the necessary setup and authorization for MongoDB Atlas to access GCP resources on behalf of your project.

## What This Example Does

> This Terraform configuration:

Reviewer 2: ```suggestion
The following Terraform configurations:
```

1. **Creates Cloud Provider Access Setup**: Establishes the initial setup for MongoDB Atlas to access GCP resources
2. **Authorizes the Access Role**: Completes the authorization process, creating a GCP service account that MongoDB Atlas can use

## Architecture Overview

> ```

Reviewer 3: nit: you can also use Mermaid diagrams, e.g. in [TF](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/network-logging.md#transport-chain) or [SDK](https://github.com/mongodb/atlas-sdk-go/blob/main/tools/README.md) , similarly for the other diagrams.

For this case I think it would be something like this:
```
graph LR
      subgraph "MongoDB Atlas"
          A[Project<br/>Cloud Provider<br/>Access]
      end

      subgraph "Google Cloud"
          B[Service Account<br/>Created by Atlas]
      end
      A <--> B
```

```mermaid
graph LR
      subgraph "MongoDB Atlas"
          A[Project<br/>Cloud Provider<br/>Access]
      end

      subgraph "Google Cloud"
          B[Service Account<br/>Created by Atlas]
      end
      A <--> B
```

Reviewer 4: Addressed in [9a8705](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3645/commits/99a87052ecb8229d82c03ebbf397d3fff641f645)

Reviewer 3: nice!  I think there is another diagram in the PR in case you also want to change it

┌─────────────────────┐    ┌─────────────────────────┐
│   MongoDB Atlas     │    │     Google Cloud        │
│                     │    │                         │
│  ┌───────────────┐  │    │                         │
│  │   Project     │  │    │                         │
│  │               │  │    │                         │
│  │ Cloud Provider│  │    │                         │
│  │    Access     │  │◄──►│   Service Account       │
│  │               │  │    │   (Created by Atlas)    │
│  │               │  │    │                         │
│  └───────────────┘  │    │                         │
│                     │    │                         │
└─────────────────────┘    └─────────────────────────┘
```

## Prerequisites

Before running this example, you need:

1. **MongoDB Atlas Account**: With API keys that have project owner permissions
2. **Google Cloud Platform Account**: With a project (no specific GCP permissions required for this setup)
3. **Terraform**: Version 0.13 or later

> ## Usage

Reviewer 2: ```suggestion
## Procedure
```

### 1. Set Up Variables

Create a `terraform.tfvars` file:

```hcl
atlas_public_key  = "<ATLAS_PUBLIC_KEY>"
atlas_private_key = "<ATLAS_PRIVATE_KEY>"
atlas_project_id  = "<ATLAS_PROJECT_ID>"
```

### 2. Deploy the Infrastructure

```bash
terraform init
terraform plan
terraform apply
```

### 3. Verify the Setup

> After successful deployment, you should see outputs including:

Reviewer 2: ```suggestion
After having successfully deployed your infrastructure, you should get the following outputs:
```
